ark/utils/spatial_analysis_utils.py

```diff
@@ -17,27 +17,39 @@ def calc_dist_matrix(label_map, path=None):
         path (str):
             path to save file. If None, then will directly return
     Returns:
+
         dict:
-            Dictionary containg cells x cells matrices with the euclidian distance between centers
-            of corresponding cells for every fov
-    """
+            Contains a cells x cells matrix with the euclidian
+            distance between centers of corresponding cells for every fov,
+            note that each distance matrix is of type xarray"""
+
     # Check that file path exists, if given
+
     if path is not None:
         if not os.path.exists(path):
-            raise ValueError("File path not valid")
+            raise FileNotFoundError("File path not valid")
+
     dist_mats_list = []
+
     # Extract list of fovs
-    fovs = list(label_map.coords['fovs'].values)
-    for i in range(0, label_map.shape[0]):
-        props = skimage.measure.regionprops(
-            label_map.loc[fovs[i], :, :, "segmentation_label"].values
-        )
-        a = [props[j].centroid for j in range(len(props))]
-        centroids = np.array(a)
+    fovs = label_map.coords['fovs'].values
```

Comment:
label map is an xarray, not a numpy array, so we should update the docstring. I also think this is a bad variable name since it's singular